Alloy/Renderers/PointsRenderer.swift

```diff
@@ -0,0 +1,143 @@
+//
+//  PointsRenderer.swift
+//  Alloy
+//
+//  Created by Eugene Bokhan on 26/04/2019.
+//
+
+import Metal
+import simd
+
+@available(iOS 11.3, tvOS 11.3, macOS 10.13, *)
+final public class PointsRenderer {
+
+    public enum Errors: Error {
+        case functionCreationFailed
+        case libraryCreationFailed
+        case wrongRenderTargetTextureUsage
+        case missingRenderTarget
+    }
+
+    // MARK: - Properties
+
+    /// Point positions described in a normalized coodrinate system.
+    public var pointsPositions: [simd_float2] {
+        set {
+            var pointsPositions = newValue
+            self.pointCount = pointsPositions.count
+            self.pointsPositionsBuffer = self.context.device
+                .makeBuffer(bytes: &pointsPositions,
+                            length: MemoryLayout<simd_float2>.stride * pointsPositions.count,
+                            options: .storageModeShared)
+        }
+        get {
+            if let pointsPositionsBuffer = self.pointsPositionsBuffer,
+                let pointsPositions = pointsPositionsBuffer
+                    .array(of: simd_float2.self,
+                           count: pointsPositionsBuffer.length / MemoryLayout<simd_float2>.stride) {
+                return pointsPositions
+            } else {
+                return []
+            }
+        }
+    }
+    /// Point color. Red in default.
+    public var color: vector_float4 = .init(1, 0, 0, 1)
+    /// Point size in pixels
+    public var pointSize: Float = 40
+
+    private var pointsPositionsBuffer: MTLBuffer?
+    private var pointCount: Int = 0
+
+    private let context: MTLContext
+    private let renderPipelineState: MTLRenderPipelineState
+
+    // MARK: - Life Cycle
+
+    /// Creates a new instance of PointsRenderer.
+    ///
+    /// - Parameter context: Alloy's Metal context.
+    /// - Throws: library or function creation errors.
+    public init(context: MTLContext, pixelFormat: MTLPixelFormat = .bgra8Unorm) throws {
+        self.context = context
+
+        guard
+            let library = context.shaderLibrary(for: Bundle(for: PointsRenderer.self))
+        else { throw Errors.libraryCreationFailed }
+
+        guard
+            let vertexFunction = library.makeFunction(name: PointsRenderer.vertexFunctionName),
+            let fragmentFunction = library.makeFunction(name: PointsRenderer.fragmentFunctionName)
+        else { throw Errors.functionCreationFailed }
+
+        let renderPipelineDescriptor = MTLRenderPipelineDescriptor()
+        renderPipelineDescriptor.vertexFunction = vertexFunction
+        renderPipelineDescriptor.fragmentFunction = fragmentFunction
+
+        if let colorAttachmentsDescriptor = renderPipelineDescriptor.colorAttachments[0] {
```

Comment:
are you sure that's adds any safeness? I think it will crash if colorAttachments is empty